apps/ensindexer/src/plugins/efp/lib/lsl-parser.ts

```diff
@@ -0,0 +1,169 @@
+   * This identifies the kind of data the data field contains.
+   */
+  type: string;
+
+  /**
+   * 32-byte EVM chain ID of the chain where the EFP list records are stored.
+   */
+  chainId: string;
+
+  /**
+   * The 20-byte EVM address of the contract where the list is stored.
+   */
+  listRecordsAddress: string;
+
+  /**
+   * The 32-byte value that specifies the storage slot of the list within the contract.
+   * This disambiguates multiple lists stored within the same contract and
+   * de-couples it from the EFP List NFT token id which is stored on Ethereum and
+   * inaccessible on L2s.
+   */
+  slot: string;
+}
+
+/**
+ * Slice encoded LSL into a dictionary of LSL params.
+ * @param encodedLsl
+ * @returns {EncodedListStorageLocation} LSL params.
+ * @throws {Error} when input param is not of expected length
+ */
+function sliceEncodedLsl(encodedLsl: string): EncodedListStorageLocation {
+  if (encodedLsl.length !== 174) {
+    throw new Error("Encoded List Storage Location values must be a 174-character long string");
+  }
+
+  return {
+    // Extract the first byte after the 0x (2 hex characters = 1 byte)
+    version: encodedLsl.slice(2, 4),
+
+    // Extract the second byte
+    type: encodedLsl.slice(4, 6),
+
+    // Extract the next 32 bytes to get the chain id
+    chainId: encodedLsl.slice(6, 70),
+
+    // Extract the address (40 hex characters = 20 bytes)
+    listRecordsAddress: encodedLsl.slice(70, 110),
+
+    // Extract last 32 bytes to get the slot
+    slot: encodedLsl.slice(110, 174),
+  } satisfies EncodedListStorageLocation;
+}
+
+/**
+ * Create a zod schema covering validations and invariants enforced with {@link parseEncodedListStorageLocation} parser.
+ * @param {number[]} efpChainIds List of IDs for chains that the EFP protocol has been present on.
+ */
+const createEfpLslSchema = (efpChainIds: number[]) =>
+  z
+    .object({
+      version: z.literal("01").transform(() => ListStorageLocationVersion.V1),
+
+      type: z.literal("01").transform(() => ListStorageLocationType.EVMContract),
+
+      chainId: z
+        .string()
+        .length(64)
+        .transform((v) => BigInt("0x" + v))
+        .refine((v) => v > BigInt(Number.MIN_SAFE_INTEGER) || v < BigInt(Number.MAX_SAFE_INTEGER), {
+          message:
+            "chainId must be a value between Number.MIN_SAFE_INTEGER and Number.MAX_SAFE_INTEGER",
+        })
+        .transform((v) => Number(v)),
+
+      listRecordsAddress: z
+        .string()
+        .length(40)
+        .transform((v) => `0x${v}`)
+        .transform((v) => getAddress(v)),
+
+      slot: z
+        .string()
+        .length(64)
+        .transform((v) => BigInt("0x" + v)),
+    })
+    // invariant: chainId is from one of the chains that EFP has been present on
+    // https://docs.efp.app/production/deployments/
+    .refine((v) => efpChainIds.includes(v.chainId), {
+      message: `chainId must be one of the EFP Chain IDs: ${efpChainIds.join(", ")}`,
```

Comment:
```suggestion
      message: `chainId must be one of the EFP Deployment Chain IDs: ${efpChainIds.join(", ")}`,
```

Ideally this message would also include a reference to the ENS Deployment Chain this statement is specific to.